Brokerages/BrokerageMultiWebSocketSubscriptionManager.cs

```diff
@@ -0,0 +1,251 @@
+        /// <param name="maximumSymbolsPerWebSocket">The maximum number of symbols per websocket connection</param>
+        /// <param name="maximumWebSocketConnections">The maximum number of websocket connections allowed (if zero, symbol weighting is disabled)</param>
+        /// <param name="connectionRateLimiter">The rate limiter for creating new websocket connections</param>
+        /// <param name="symbolWeights">A dictionary for the symbol weights</param>
+        /// <param name="webSocketFactory">A function which returns a new websocket instance</param>
+        /// <param name="subscribeFunc">A function which subscribes a symbol</param>
+        /// <param name="unsubscribeFunc">A function which unsubscribes a symbol</param>
+        /// <param name="messageHandler">The websocket message handler</param>
+        public BrokerageMultiWebSocketSubscriptionManager(
+            string webSocketUrl,
+            int maximumSymbolsPerWebSocket,
+            int maximumWebSocketConnections,
+            Dictionary<Symbol, int> symbolWeights,
+            Func<WebSocketClientWrapper> webSocketFactory,
+            Action<IWebSocket, Symbol> subscribeFunc,
+            Action<IWebSocket, Symbol> unsubscribeFunc,
+            BrokerageConcurrentMessageHandler<WebSocketMessage> messageHandler,
+            RateGate connectionRateLimiter = null)
+        {
+            _webSocketUrl = webSocketUrl;
+            _maximumSymbolsPerWebSocket = maximumSymbolsPerWebSocket;
+            _maximumWebSocketConnections = maximumWebSocketConnections;
+            _webSocketFactory = webSocketFactory;
+            _subscribeFunc = subscribeFunc;
+            _unsubscribeFunc = unsubscribeFunc;
+            _messageHandler = messageHandler;
+            _connectionRateLimiter = connectionRateLimiter;
+
+            if (_maximumWebSocketConnections > 0)
+            {
+                // symbol weighting enabled, create all websocket instances
+                for (var i = 0; i < _maximumWebSocketConnections; i++)
+                {
+                    var webSocket = _webSocketFactory();
+                    webSocket.Open += OnOpen;
+
+                    _webSocketEntries.Add(new BrokerageMultiWebSocketEntry(symbolWeights, webSocket));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Subscribes to the symbols
+        /// </summary>
+        /// <param name="symbols">Symbols to subscribe</param>
+        /// <param name="tickType">Type of tick data</param>
+        protected override bool Subscribe(IEnumerable<Symbol> symbols, TickType tickType)
+        {
+            Log.Trace($"BrokerageMultiWebSocketSubscriptionManager.Subscribe(): {string.Join(",", symbols.Select(x => x.Value))}");
+
+            foreach (var symbol in symbols)
+            {
+                var webSocket = GetWebSocketForSymbol(symbol);
+
+                _subscribeFunc(webSocket, symbol);
```

Comment:
We should make these guy `_subscribeFunc` `_unsubscribeFunc` return a `bool` and use it to determine the result of this method, not hard coded `true` response 👍